src/content/url-handling.js

```diff
@@ -50,7 +50,10 @@ export function urlFromState(urlState) {
     invertCallstack: urlState.invertCallstack ? null : undefined,
     jsOnly: urlState.jsOnly ? null : undefined,
     callTreeFilters: stringifyCallTreeFilters(urlState.callTreeFilters[urlState.selectedThread]) || undefined,
-  } : {});
+  } : (urlState.selectedTab === 'flameChart' ? {
```

Comment:
I'm having a lot of trouble unpacking all of this syntax with ternaries and inline code. Would you mind splitting this out a bit, or reformatting it to make it easier to read?

Not sure you have an opinion here, but my preferred method for multiline ternaries is the following form, which may or may not help in this case...

```
condition
  ? option1
  : option2
```